src/Microsoft.OpenApi/Extensions/EnumExtensions.cs

```diff
@@ -37,10 +40,96 @@ public static class EnumExtensions
         /// Use <see cref="DisplayAttribute"/> if exists.
         /// Otherwise, use the standard string representation.
         /// </returns>
+        [Obsolete("Use native AoT-friendly type-specific overloads GetDisplayName methods instead.")]
+        [RequiresUnreferencedCode("GetAttributeOfType is not trim-compatible. Recommended to use native AoT-friendly type-specific overloads of GetDisplayName instead.")]
         public static string GetDisplayName(this Enum enumValue)
         {
             var attribute = enumValue.GetAttributeOfType<DisplayAttribute>();
             return attribute == null ? enumValue.ToString() : attribute.Name;
         }
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="ParameterStyle" /> without the use of reflection.
+        /// </summary>
+        /// <param name="parameterStyle">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        internal static string GetDisplayName(this ParameterStyle parameterStyle) => parameterStyle switch
+            {
+                ParameterStyle.Matrix => "matrix",
+                ParameterStyle.Label => "label",
+                ParameterStyle.Form => "form",
+                ParameterStyle.Simple => "simple",
+                ParameterStyle.SpaceDelimited => "spaceDelimited",
+                ParameterStyle.PipeDelimited => "pipeDelimited",
+                ParameterStyle.DeepObject => "deepObject",
+                _ => throw new InvalidOperationException($"Unknown parameter style: {parameterStyle}")
+            };
+
+        /// <summary>
+        /// Gets the enum display for name <see cref="ParameterLocation" /> without the use of reflection. 
+        /// </summary>
+        /// <param name="parameterLocation">The enum value.</param>
+        /// <returns>The display string to use.</returns>
+        public static string GetDisplayName(this ParameterLocation parameterLocation) => parameterLocation switch
```

Comment:
Why are some of these methods internal, and some public?

Reply:
We should have an "InternalsVisibleTo" attribute somewhere already making this method visible to the other assembly.

Reply:
We should have an "InternalsVisibleTo" attribute somewhere already making this method visible to the other assembly.

Reply:
The typical stance I've used for using IVT between product code is if they "ship together" and will never have mismatched versions. If 2 libraries ship in separate NuGet packages, I don't use IVT between them. One could get updated without the other, and if the internal API changed the assemblies won't load correctly.

Reply:
I don't have a strong opinion either way (besides consistency between the methods maybe). I think the other rule of thumb I'd use in this scenario is: do we expect people to need this at any point? I don't think anybody is going to re-implement serialization and deserialization of OAI specifications in dotnet when they already have a reference to OAI.net. (maybe a different format, if even?) so using IVT should be fine here IMHO.

Reply:
I'd be curious to see what @MaggieKimani1 and @darrelmiller are thinking here.

I'm personally starting to lean towards public API over IVT for the reasons that @eerhardt mentioned and the fact that that seems to be the pre-existing convention for this library (as annoying it is to have to make a single API public).

Reply:
Note that if we do what's outlined in https://github.com/microsoft/OpenAPI.NET/pull/1717#discussion_r1681224383, we can get rid of these new methods.

Reply:
> Note that if we do what's outlined in [#1717 (comment)](https://github.com/microsoft/OpenAPI.NET/pull/1717#discussion_r1681224383), we can get rid of these new methods.

Based on https://github.com/dotnet/runtime/issues/97737#issuecomment-2233902387, it looks like we'll have to wait for a follow-up change to the runtime to support this. It seems like this should happen before preview.7 so I am fine waiting.

Reply:
OK -- took me a while to catch up with the fact that this code was always trim-safe despite the warnings 😅 

I think https://github.com/microsoft/OpenAPI.NET/pull/1717/commits/c117bc56f17317853763b065aac943a1eedee9f7 does the right thing here.